src/main/resources/Macros/KanbanAWMSource.xml

```diff
@@ -119,17 +119,38 @@
   #if($request.debug)
   /* {{{ $results }}} */
   #end
-  #if ($categoryValues)
-    #set ($categories = $categoryValues)
-  #else
-    #set ($categoryField = $xwiki.getDocument($className).getxWikiClass().get($category))
-    #set ($categories = $categoryField.getListValues())
+  #set ($categoryField = $xwiki.getDocument($className).getxWikiClass().get($category))
+  #set ($categories = $categoryField.getListValues())
+
+  #if ($selectedCategoryValues)
+    ## Validate the requested category values.
+    #if ($categories)
+      ## Remove values not in the set of possible values (if there is such a set).
+      #set ($validatedCategoryValues = [])
+      #foreach ($selectedCategoryValue in $selectedCategoryValues)
+        #if ($categories.contains($selectedCategoryValue))
```

Comment:
You could include the 2 conditions in here
`#if ($categories.contains($selectedCategoryValue) && !$validatedCategoryValues.contains($selectedCategoryValue))`
which means that you add a value only if it's valid and it wasn't added already

The other option, which is even simpler, is to iterate through $categories and only add a value to `validCategories` if `selectedCategoryValue` contains it

Reply:
Is there any reason why values may be missing from the ```$categoryField.getListValues()```? There are these lines below:
```velocity
    #if(!$categories.contains($value) && !$selectedCategoryValues)
     #set($discard = $categories.add($value))
    #end
```
They still trigger when no ```displayedCategoryValues``` parameter is present, but it means that in the full view there may be certain values which are not accessible in the parameter. These might be invalid by design, but I'm not sure if there exists an use case when this is desired.

Example: (selecting the ```bold``` status is treated like an invalid value, which it is, but the user asked for it directly so it might be surprising to the user that it isn't shown)
![image](https://github.com/user-attachments/assets/37da95f2-80db-4eda-a29e-baa1479d0b75)